napari/_qt/_qapp_model/_tests/test_file_menu.py

```diff
@@ -148,6 +149,108 @@ def test_show_shortcuts_actions(make_napari_viewer):
     viewer.window._pref_dialog.close()
 
 
+def test_image_from_clipboard(make_napari_viewer):
+    viewer = make_napari_viewer()
+
+    # Ensure clipboard is empty
+    QGuiApplication.clipboard().clear()
+    clipboard_image = QGuiApplication.clipboard().image()
+    assert clipboard_image.isNull()
+
+    # Check action via trigger
+    actions = viewer.window.file_menu.actions()
+    action = None
+    for from_clipboard_action in actions:
+        if from_clipboard_action.text() == 'New Image from Clipboard':
+            action = from_clipboard_action
+            break
+    assert action
+    with mock.patch('napari._qt.qt_viewer.show_info') as mock_show_info:
+        action.trigger()
+    mock_show_info.assert_called_once_with('No image or link in clipboard.')
+
+    # Check action via execute_command
```

Comment:
My understanding is that `execute_command` is the app-model way to call an action callback while `trigger` is a way to activate a `QAction`. Both calls should end up calling the same logic defined as an app-model `Action` callback but one does it via the app-model command registry ( `app_model.registries.CommandRegistry.execute_command`) while the other uses the actual GUI machinery (`qtpy.QtWidgets.QAction.trigger` in this case since Qt is being used).

Let me know if that makes sense or maybe I'm missing something and testing just one method is enough!

Reply:
I think this is nice to have, because it could lead to easier debugging based on which is failing (or if both are failing). Happy to keep both!

Reply:
Okay we dug into this a bit more and it seems that `QCommandAction` calls `execute_command` when it is triggered, see: https://github.com/pyapp-kit/app-model/blob/main/src/app_model/backends/qt/_qaction.py#L55

So we think `trigger` and `execute_command` are effectively the same. I've taken the read-to-merge label off, but please let us know what you think @dalthviz

Reply:
Oh I see, thanks for the check! I will change things here to only call `trigger` then :+1:

Reply:
I would almost use `execute_command` because then you don't have to do `findAction` first, but up to you.